app/code/Magento/Inventory/Model/ReservationRepository.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Inventory\Model\Reservation\Command\GetListInterface;
+use Magento\InventoryApi\Api\ReservationRepositoryInterface;
+
+/**
+ * @inheritdoc
+ */
+class ReservationRepository implements ReservationRepositoryInterface
```

Comment:
@aleron75  you are right. Regarding the API and SPI, and that Repository should play the role of API.

From my point of view,  @naydav meant that we already have an API for Reservation Retrieving: _GetReservationsForProductInterface_ 
And we don't have another Business Use-Cases for Reservation Search (with Filtering) where Repository::getList would be needed for us.

That's why business logic  **would not** use ReservationRepository as an API (Business Logic will not work with ReservationRepository directly, calling its methods).
It will always work indirectly through the  GetReservationsForProductInterface API call.
Thus, in this case, ReservationRepository become an SPI, but not API.

But that's not a desirable state for us, as it overcomplicates object dependencies:
GetReservationsForProductInterface (API) -> ReservationRepository (SPI) -> GetListCommandInterface (SPI)
Too many extension points